go-controller/pkg/node/ovspinning/ovspinning_linux.go

```diff
@@ -0,0 +1,179 @@
+
+	klog.Infof("Starting OVS daemon CPU pinning")
+
+	fileWatcher, err := createFileWatcherFor(featureEnablerFile)
+	if err != nil {
+		klog.Warningf("Can't start OVS CPU affinity pinning: %v", err)
+		return
+	}
+	ticker := time.NewTicker(tickDuration)
+
+	// If the file exists and is not empty, enable the affinity pinning
+	isFeatureEnabled, err := isFileNotEmpty(featureEnablerFile)
+	if err != nil {
+		klog.Warningf("Error while reading [%s]: %v", featureEnablerFile, err)
+		isFeatureEnabled = false
+	}
+
+	for {
+		select {
+		case _, ok := <-fileWatcher.Events:
+			if !ok {
+				continue
+			}
+
+			isFeatureEnabled, err = isFileNotEmpty(featureEnablerFile)
+			if err != nil {
+				klog.Warningf("Error while reading [%s]: %v", featureEnablerFile, err)
+				continue
+			}
+
+		case err, ok := <-fileWatcher.Errors:
+			if ok {
+				klog.Errorf("Error watching for file changes: %s", err)
+			}
+
+		case <-stopCh:
+			klog.Infof("Stopping OVS daemon CPU pinning")
+			return
+
+		case <-ticker.C:
+			if !isFeatureEnabled {
+				continue
+			}
+
+			err := alignOvsDBServer()
+			if err != nil {
+				klog.Warningf("Error while aligning ovsdb-server CPUs to current process: %v", err)
+			}
+
+			err = alignOvsVSwitchd()
+			if err != nil {
+				klog.Warningf("Error while aligning ovs-vswitchd CPUs to current process: %v", err)
+			}
+		}
+	}
+}
+
+func createFileWatcherFor(filename string) (*fsnotify.Watcher, error) {
+	fileWatcher, err := fsnotify.NewWatcher()
+	if err != nil {
+
+		return nil, fmt.Errorf("failed to create filesystem watcher: %w", err)
+	}
+
+	err = fileWatcher.Add(featureEnablerFile)
+	if err != nil {
+		return nil, fmt.Errorf("unable to watch [%s] file: %w", filename, err)
+	}
+
+	return fileWatcher, nil
+}
+
+func isFileNotEmpty(filename string) (bool, error) {
+
+	readFile, err := os.Open(filename)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
+		}
+
+		return false, fmt.Errorf("can't open enabler file [%s]: %w", filename, err)
+	}
+
+	b := make([]byte, 1)
+	n, err := readFile.Read(b)
+	if err != nil {
+		if err == io.EOF {
+			// The file is empty
+			return false, nil
+		}
+
+		return false, fmt.Errorf("can't read enabler file [%s]: %w", filename, err)
+	}
+
+	return n > 0, nil
+}
+
+func alignOvsVSwitchd() error {
+	klog.V(5).Info("Aligning ovs-vswitchd daemon CPU affinity with the ones from the current process")
+
+	ovsVSwitchdPID, err := getOvsVSwitchdPIDFn()
+	if err != nil {
+		return fmt.Errorf("can't retrieve ovs-vswitchd PID: %w", err)
+	}
+
+	return alignProcessCPUWithCurrentOnes(ovsVSwitchdPID)
+}
+
+func alignOvsDBServer() error {
+	klog.V(5).Info("Aligning ovs-vswitchd daemon CPU affinity with the ones from the current process")
+
+	ovsDBserverPID, err := getOvsDBServerPIDFn()
+	if err != nil {
+		return fmt.Errorf("can't retrieve ovsdb-server PID: %w", err)
+	}
+
+	return alignProcessCPUWithCurrentOnes(ovsDBserverPID)
+}
+
+func alignProcessCPUWithCurrentOnes(targetPIDStr string) error {
```

Comment:
maybe rename to "setProcessCPUAffinity" and put doc text on the function explaining that it sets the given process pid to the same affinity as ovnkube-node